field/eisenstein/eisenstein_test.go

```diff
@@ -273,6 +272,6 @@ func BenchmarkHalfGCD(b *testing.B) {
 	c := ComplexNumber{A0: *c0, A1: *c1}
 	b.ResetTimer()
 	for i := 0; i < b.N; i++ {
-		HalfGCD(&a, &c)
+		_ = HalfGCD(&a, &c)
```

Comment:
I'm not sure it is sufficient. I think it is better to create a file level variable `var benchRet [3]*ComplexNumber` and then assign to it

From Dave Cheney's blog:
```go
var result int

func BenchmarkFibComplete(b *testing.B) {
        var r int
        for n := 0; n < b.N; n++ {
                // always record the result of Fib to prevent
                // the compiler eliminating the function call.
                r = Fib(10)
        }
        // always store the result to a package level variable
        // so the compiler cannot eliminate the Benchmark itself.
        result = r
}
```